libs/shared/methodologies/bold/getters/src/document.getters.ts

```diff
@@ -9,43 +10,47 @@ import {
   DocumentEventName,
   MassIdDocumentActorType,
 } from '@carrot-fndn/shared/methodologies/bold/types';
+import { getYear } from 'date-fns';
 
 import { getEventAttributeValue } from './event.getters';
 
 const { DROP_OFF, EMISSION_AND_COMPOSTING_METRICS, PICK_UP, RULES_METADATA } =
   DocumentEventName;
 const { PROCESSOR, RECYCLER, WASTE_GENERATOR } = MassIdDocumentActorType;
 
-export const getLastEmissionAndCompostingMetricsEvent = (
-  document: Document,
-): DocumentEvent | undefined => {
-  const emissionAndCompostingMetricsEvents = document.externalEvents?.filter(
-    (event) => event.name.toString().includes(EMISSION_AND_COMPOSTING_METRICS),
+interface LastYearEmissionAndCompostingMetricsEventParameters {
+  documentWithEmissionAndCompostingMetricsEvent: Document;
+  documentWithYear: Document;
```

Comment:
In my opinion, it would be clearer if we already pass the year. What do you think?
```suggestion
  currentYear: NonZeroPositiveInt;
```

Then we can have another helper such as extractYear and call it with:
```
  currentYear: extractYear(massId.externalCreatedAt)
```